### PR TITLE
Complete Weather APP Reactjs

### PR DESCRIPTION
**Pull request by Kanyorok**
The completed tasks were:

- Used React documentation.
- Used React components.
- Used React props.
- Used React Router.
- Connected React and Redux.
- Handled events in a React app.
- Wrote integration tests with a React testing library.
- Used styles in a React app.
- Used React life cycle methods.
- Applied React best practices and language style guides in code.
- Used store, actions and reducers in React.

### CODE REVIEW

File: src/components/Aqidata.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import millify from 'millify';
+import Header from './Header';
+import { displayWeatherData } from '../redux/Aqidata/aqidataSlice';
+
+const Aqiweather = () => {
+  const location = useLocation();
+  const { city } = location.state;
```
Comment (Davidosky007) [inline src/components/Aqidata.jsx:10]: * I get errors on my console when i try to run your application , kindly work on it 
image

File: README.md [hunk 2]
```diff
@@ -1,2 +1,175 @@
...
+- **Search and filter cities**
+- **View population of people in a city**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://climaticvals.onrender.com/)
```
Comment (Davidosky007) [inline README.md]: * The live demo link doesn't work as expected, kindly ensure the home and details page works

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Kanyorok 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* All tests passed 👍 
* Redux setup 👍 
* Well written PR 👍 

## Required Changes ♻️
```
* Kindly ensure you recorded a video presenting the features of the project.
The maximum length of the video is 5 minutes.
The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
The video contains a demo of the project, highlighting all the project features.
The video contains a highlight about a piece of the code the student finds interesting or is proud of making.
```
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kanyorok** [comment]
Hello @Davidosky007,

Thank you for the review. I have made changes to the live demo and readme video. However, I have a question about [location error](https://github.com/Kanyorok/air-pollution-aqi/pull/1#discussion_r1305165558). Please, could you let me call you on slack you show it to me? It is not showing on my side. So I want to only confirm why it is showing as so on your side.

File: src/components/Aqidata.jsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+              </td>
+            </tr>
+            <tr className="bg-[#e94988] text-white">
+              <td className="w-1/6 whitespace-wrap px-2 py-2">
+                <div className="flex items-start justify-between">
+                  <h2 className="font-bold">CO</h2>
+                  <h3>{weatherData.components.co}</h3>
+                </div>
+              </td>
+            </tr>
```
Comment (Strangeal) [inline src/components/Aqidata.jsx:71]: - [ ] Great job so far implementing all the changes suggested by the previous reviewer. However, there are still changes you need to make in your design to improve it and make it match that of the given design in the project requirements. To do so kindly consider the below suggestions:

    - [ ] Upon comparing your design with that of the given one I noticed from the homepage that you're missing an `image`. On the given design each card on the homepage contains `images`, `text`, and `icon` but on your design, your homepage contains `text` and `icon` with no `image`. Kindly consider adding an image on each card on your homepage. See below for more details:
 
 <table>
<tr>
<th align=“middle”>Behance design</th>
<th align=“middle”>Your design</th>
</tr>
<tr>
<td>
Screenshot 2023-08-25 at 08 46 05
</td>
<td>
Screenshot 2023-08-25 at 08 49 16
</td>
</tr>
</table>

Comment (Strangeal) [inline src/components/Aqidata.jsx:71]: - [ ] Looking at your cards you would notice that you have the details of each card displayed in its own container as compared to that of the given design. In your card, you have `country`, `city`, `population`, and `icon` displayed in a `row` when they should be displayed in a `column`.

- [ ] Kindly make sure that your card `icon` is displayed absolutely at the top of your card to match that of the given design. To have a more visual understanding of the changes kindly take a look at the screenshot below for more details:

<table>
<tr>
<th align=“middle”>Behance design</th>
<th align=“middle”>Your design</th>
</tr>
<tr>
<td>
Screenshot 2023-08-25 at 09 17 19
</td>
<td>
Screenshot 2023-08-25 at 09 16 43
</td>
</tr>
</table>

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @Kanyorok, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Alt Text

 ## Highlights
Live link ✅
Video presentation ✅


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by muneebulrehman — APPROVED** [review]
Hi @Kanyorok , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- Good follow-up with the previous reviewers.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**